embedded/store/indexer.go

```diff
@@ -312,6 +312,7 @@ func (idx *indexer) doIndexing(cancellation <-chan struct{}) {
 		idx.stateCond.L.Lock()
 		for {
 			if idx.state == stopped {
+				idx.stateCond.L.Unlock()
```

Comment:
@jeroiraz I believe this unlock is missing here, can you confirm?

Reply:
yes, for sake of completeness but it shouldn't affect any functionality